### ISSUE TITLE
Is this still the recommended way to use mongo with sourced?

### ISSUE THREAD

**Issue by chill-cod3r** [mateodelnorte/sourced-repo-mongo#49]
I had a couple of questions if someone has a chance :) 

- is this still the recommended way to use mongo as a db store with sourced package?
- what is some of the intention behind saving actual snapshots vs only saving events? I notice it only seems to create a snapshot for me (i've only been playing with a small example) if i call the method `_commitSnapshots` - otherwise it seems to be only creating events. is that right? if so, is there a way to control how often snapshots get created? is there a reason you'd want to? if so, what is an example of why you'd want to control that more granularly. In my head i would have thought creating snapshots frequently is helpful but this concept of event sourced is new to me :)
- use case wise - say a topic consumer handler function (like from your package `servicebus`) - would you typically be just saving 1 event at a time and then also build api routes around accessing your data? and when you access your data you would use this repo to handle the db connection and entity wrapping?

Thanks for contributing this and I look forward to learning more about it.

**Comment by chill-cod3r** [comment]
Hmmm, I'm not sure what I was doing before, but now when I `.commit` the entity it is creating snapshots correctly - I noticed from the repo code the default is every 10 events.

fwiw - i see a lot of good examples using this technique in your talk - https://www.youtube.com/watch?v=4k7bLtqXb8c

TL;DR - After more research I answered my own questions :)

**Comment by mateodelnorte** [comment]
Cheers!
